syncer/samples/multi-servicecenters/servicecenter/hello-server/servicecenter/servicecenter.go

```diff
@@ -0,0 +1,244 @@
+	time.Sleep(time.Second * 3)
+	// 注销微服务
+	return cli.DeleteService(ctx, domainProject, service.ServiceId)
+}
+
+func Do(ctx context.Context, method, addr string, headers http.Header, body []byte) (resp *http.Response, err error) {
+	raw, err := url.Parse(addr)
+	if err != nil {
+		return nil, err
+	}
+	endpoints, err := serverNameToEndpoints(raw.Hostname())
+
+	client, err := sc.NewLBClient(endpoints, (&sc.Config{Endpoints: endpoints}).Merge())
+	if err != nil {
+		return nil, err
+	}
+
+	return client.RestDoWithContext(ctx, method, raw.Path, headers, body)
+}
+
+func createService(ctx context.Context, svc *MicroService) (*proto.MicroService, error) {
+	service := transformMicroService(svc)
+
+	// 检测微服务是否存在
+	serviceID, err := cli.ServiceExistence(ctx, domainProject, service.AppId, service.ServiceName, service.Version, "")
+	if serviceID == "" {
+		// 注册微服务
+		serviceID, err = cli.CreateService(ctx, domainProject, service)
+		if err != nil {
+			return nil, err
+		}
+	}
+	service.ServiceId = serviceID
+	return service, nil
+}
+
+func registerInstance(ctx context.Context, serviceID string, ins *Instance) (*proto.MicroServiceInstance, error) {
+	instance := transformInstance(ins)
+	instanceID, err := cli.RegisterInstance(ctx, domainProject, serviceID, instance)
+	if err != nil {
+		return nil, err
+	}
+	instance.ServiceId = serviceID
+	instance.InstanceId = instanceID
+	return instance, nil
+}
+
+func heartbeat(ctx context.Context, ins *proto.MicroServiceInstance) {
+	// 启动定时器：间隔30s
+	t := ticker.NewTaskTicker(heartbeatInterval, func(ctx context.Context) {
+		// 定时发送心跳
+		err := cli.Heartbeat(ctx, domainProject, ins.ServiceId, ins.InstanceId)
+		if err != nil {
+			log.Errorf(err, "send heartbeat failed, domain = %s, serviceID = %s, instanceID = %s", domainProject, ins.ServiceId, ins.InstanceId)
+			return
+		}
+		log.Debug("send heartbeat success")
+	})
+	defer t.Stop()
+	t.Start(ctx)
+}
+
+func discoveryToCaches(ctx context.Context, consumerID string, provider *MicroService) error {
+	service := transformMicroService(provider)
+	list, err := cli.DiscoveryInstances(ctx, domainProject, consumerID, service.AppId, service.ServiceName, service.Version)
+	if err != nil || len(list) == 0 {
+		return fmt.Errorf("provider not found, serviceName: %s appID: %s, version: %s",
+			provider.Name, provider.AppID, provider.Version)
+	}
+	// 缓存 provider 实例信息
+	providerCaches.Store(provider.Name, list)
+	provider.ID = list[0].ServiceId
+	return nil
+}
+
+func watchAndRenewCaches(ctx context.Context, provider *MicroService) {
+	err := cli.Watch(ctx, domainProject, provider.ID, func(result *proto.WatchInstanceResponse) {
+		log.Debug("reply from watch service")
+		list, ok := providerCaches.Load(result.Instance.ServiceId)
+		if !ok {
+			log.Infof("provider \"%s\" not found", result.Instance.ServiceId)
+			return
+		}
+		providerList := list.([]*proto.MicroServiceInstance)
+
+		renew := false
+		for i, l := 0, len(providerList); i < l; i++ {
```

Comment:
4 level code block, plz simplify it

Reply:
I was changed it, plz review again, thx!

Reply:
@tianxiaoliang